### PR TITLE
Implemented Ajax planel, closes #261

### LINKED ISSUE #261: RFC - toolbar should reload periodically or on Ajax complete
Given that firelog is gone, the toolbar should listen for sent Ajax requests or reload itself periodically in order to show data collected from xhr requests.

### PR DESCRIPTION
**Pull request by lorenzo**
This is implements the Ajax panel proposed in #261 

ajax_panel

### CODE REVIEW

**Comment by zoghal** [comment]
:100: :+1:

**Comment by jadb** [comment]
:+1:

**Comment by markstory** [comment]
Clever stuff.

**Comment by markstory** [comment]
Looks great, would the interface be simpler/clearer if instead of going into a separate a separate panel Ajax requests appended to the top of the history panel? They could be flagged as xhr requests too if visually separating them is important.

File: webroot/js/toolbar-app.js [hunk 1]
```diff
@@ -221,10 +222,21 @@ Toolbar.prototype = {
 		}
 	},
 
+	onMessage: function(event) {
+		if (event.data.startsWith('ajax-completed$$')) {
+			this.ajaxRequests.push(JSON.parse(event.data.split('$$')[1]));
+		}
+    },
```
Comment (ADmad) [inline webroot/js/toolbar-app.js]: Indentation seems off.

**Comment by ADmad** [comment]
Reusing history panel and color coding ajax request would be nice indeed.

**Comment by jameswatts** [comment]
Very nice!

**Comment by lorenzo** [comment]
@markstory I would personally would find it more difficult to understand finding ajax requests in history. They would also be a bit more difficult to separate visually and to prevent from overflowing the history panel.

I'm open to ideas on how to present them visually if we are to put them in the history panel...

**Comment by lorenzo** [comment]
Added the `Content-Type` to the request information and now request are listed showing newest first

**Comment by lorenzo** [comment]
@markstory @ADmad This is what I came up with for merging the two panels
ajax panel

File: webroot/js/toolbar-app.js [hunk 2]
```diff
@@ -221,10 +222,26 @@ Toolbar.prototype = {
 		}
 	},
 
+	onMessage: function(event) {
+		if (event.data.startsWith('ajax-completed$$')) {
```
Comment (markstory) [inline webroot/js/toolbar-app.js]: I don't think startsWith is available everywhere yet.

**Comment by markstory** [comment]
@lorenzo Looks fantastic!

**Comment by ADmad** [comment]
:clap:

**Comment by lorenzo** [comment]
Going to merge and tag a new release, thanks for your feedback :)

**Pull request merged**